server/src/main/java/org/elasticsearch/search/aggregations/metrics/tophits/TopHitsAggregationBuilder.java

```diff
@@ -603,10 +597,17 @@ protected XContentBuilder internalXContent(XContentBuilder builder, Params param
         if (storedFieldsContext != null) {
             storedFieldsContext.toXContent(SearchSourceBuilder.STORED_FIELDS_FIELD.getPreferredName(), builder);
         }
-        if (fieldDataFields != null) {
+        if (docValueFields != null) {
             builder.startArray(SearchSourceBuilder.DOCVALUE_FIELDS_FIELD.getPreferredName());
-            for (String fieldDataField : fieldDataFields) {
-                builder.value(fieldDataField);
+            for (FieldAndFormat dvField : docValueFields) {
+                if (dvField.format == null) {
+                    builder.value(dvField.field);
```

Comment:
I wonder if we should no always render this with the expanded object form to be explicit?

Reply:
Fine with me. Should we skip the `format` when it is null or emit an explicit `null`?

Reply:
I would skip format when its null but always use the object variant